### PR TITLE
Changed RPC port option selection flow in the installer

### PR DESCRIPTION
**Pull request by jpop32**
Changes to make implications of RPC port selection more explicit.

### CODE REVIEW

**Review by yulin-dong — COMMENTED** [review]
Any documents should be updated with this change? @jpop32

**Comment by jpop32** [comment]
> Any documents should be updated with this change?

Yes, [this one](https://docs.avax.network/nodes/build/set-up-node-with-installer), mostly just to reflect the difference in messages. Once we're ok with the changes in the script, I'll update the tutorial.

**Comment by yulin-dong** [comment]
> > Any documents should be updated with this change?
> 
> Yes, [this one](https://docs.avax.network/nodes/build/set-up-node-with-installer), mostly just to reflect the difference in messages. Once we're ok with the changes in the script, I'll update the tutorial.

I am good with the script change. Please update the doc in the same PR so that they are consistent.

**Review by cgcardona — COMMENTED** [review]
LGTM once the associated tutorial is updated ✅

File: scripts/avalanchego-installer.sh [hunk 1]
```diff
@@ -366,14 +366,29 @@ else
...
-echo "Your node can accept RPC calls on port 9650. If restricted to local, it will be accessible only from this machine."
-while [ "$rpcOpt" != "any" ] && [ "$rpcOpt" != "local" ]
+echo "Your node accepts RPC calls on port 9650. If restricted to private, it will be accessible only from this machine."
+echo "Only p2p port (9651 by default) NEEDS to be publicly accessible for correct node operation, RPC port is used for"
+echo "interaction with the node by the operator or applications and SHOULD NOT be freely accessible to the public."
+echo ""
+while [ "$rpcOpt" != "public" ] && [ "$rpcOpt" != "private" ]
 do
-  read -p "Do you want the RPC port to be accessible to any or only local network interface? [any, local]: " rpcOpt
+  read -p "Do you want the RPC port to be accessible to public or private (local) network interface? [public, private]: " rpcOpt
```
Comment (patrick-ogrady) [inline scripts/avalanchego-installer.sh:375]: We should explicitly add here that the recommended for a validator is private.

Comment (jpop32) [reply scripts/avalanchego-installer.sh:375]: It is mentioned in the confirmation text, but I'll pull it up to here.

**Review by yulin-dong — APPROVED** [review]

**Pull request merged**